cmd/helm/repo_remove_test.go

```diff
@@ -95,4 +81,74 @@ func TestRepoRemove(t *testing.T) {
 	if f.Has(testRepoName) {
 		t.Errorf("%s was not successfully removed from repositories list", testRepoName)
 	}
+
+	// Test removal of multiple repos in one go
+	var testRepoNames = []string{"foo", "bar", "baz"}
+	b2 := bytes.NewBuffer(nil)
+	idxs := make(map[string][]string, len(testRepoNames))
+
+	for _, repoName := range testRepoNames {
+		o := &repoAddOptions{
+			name:     repoName,
+			url:      ts.URL(),
+			repoFile: repoFile,
+		}
+
+		if err := o.run(os.Stderr); err != nil {
+			t.Error(err)
+		}
+
+		cacheIndex, cacheChart := indexing(b, rootDir, repoName)
+		idxs[repoName] = []string{cacheIndex, cacheChart}
+
+	}
+
+	multiRmOpts := repoRemoveOptions{
+		names:     testRepoNames,
+		repoFile:  repoFile,
+		repoCache: rootDir,
+	}
+
+	if err := multiRmOpts.run(b2); err != nil {
+		t.Errorf("Error removing list of repos from repositories: %q", testRepoNames)
+	}
+	if !strings.Contains(b2.String(), "has been removed") {
+		t.Errorf("Unexpected output: %s", b.String())
+	}
+
+	for _, repoName := range testRepoNames {
+		f, err := repo.LoadFile(repoFile)
+		if err != nil {
+			t.Error(err)
+		}
+		if f.Has(repoName) {
+			t.Errorf("%s was not successfully removed from repositories list", repoName)
+		}
+		cacheIndex := idxs[repoName][0]
+		cacheChart := idxs[repoName][1]
+		testIndices(t, cacheIndex, cacheChart, repoName)
```

Comment:
I'm pretty unhappy about this solution. Looks pretty messy, and the definition of `idxs` on line 102 is pretty bad too. Do we even need to check each repo individually, or can we perhaps check them all in one swoop, getting rid of having to store the index files for each repo...?

Reply:
Personally, I would go simpler.  You are creating two repos, and each repo will have an index file and a cache file.  The test just needs to make sure both files are removed after the repos are removed.  I would just have 4 variables. Instead of the old `idx` and `idx2` which were poorly named anyway, I would simply have  `indexFile1`, `cacheFile1` and `indexFile2`, `cacheFile2`.